src/project.ts

```diff
@@ -132,6 +143,62 @@ export async function readProject(
   };
 }
 
+/**
+ * Returns the https-prefixed GitHub repository URL for the project, so that we
+ * know how to construct links to releases inside of the changelog.
+ *
+ * This URL is obtained by looking at the following sources, in this order:
+ *
+ * - `process.env.npm_package_repository_url` (to support MetaMask projects
+ * which still use NPM or Yarn 1.x)
+ * - The `repository` field inside of `package.json` (to support most MetaMask
+ * projects)
+ * - The URL of the "origin" remote (to support newer projects which for some
+ * reason do not have a "repository" field)
+ *
+ * Note that there is a function in `@metamask/auto-changelog` that offers a
+ * similar capability, and in fact, this function should match it as closely as
+ * possible so that when changelogs are updated they also pass
+ * `@metamask/auto-changelog`'s validation step.
+ *
+ * @param packageManifest - The manifest for the package that represents a
+ * project.
+ * @param repositoryDirectoryPath - The path to the repository for the project.
+ * @returns The HTTPS URL of the repository, e.g.
+ * `https://github.com/OrganizationName/RepositoryName`.
+ * @throws If the URL obtained via `package.json` or the "origin" remote is in
+ * in invalid format.
+ */
+export async function getValidRepositoryUrl(
+  packageManifest: UnvalidatedPackageManifest,
+  repositoryDirectoryPath: string,
+): Promise<string> {
+  // Set automatically by NPM or Yarn 1.x
```

Comment:
Although handling NPM or Yarn 1.x projects is not strictly necessary here, because `create-release-branch` only supports monorepos, and all of our monorepos use Yarn Modern, I figured I'd include it here because I want to copy this function back to `auto-changelog` after this PR is merged.

For context the original source for this logic is here: https://github.com/MetaMask/auto-changelog/blob/ef3e86e15b0de7061856a53fd18c4f38e898f5e8/src/repo.ts#L20